features/steps/givens/relationships.py

```diff
@@ -61,7 +62,20 @@ def make_aggregate(val):
     if not instances and required:
         yield ValidationOutcome(instance_id=inst, severity=OutcomeSeverity.ERROR)
 
-    
+
+@gherkin_ifc.step("Each relating material layer")
+def step_impl(context, inst):
+    if inst.is_a("IfcMaterialLayerSet"):
```

Comment:
You should also take into account the _Usage brothers of these material aggregates.

https://ifc43-docs.standards.buildingsmart.org/IFC/RELEASE/IFC4x3/HTML/lexical/IfcMaterialLayerSetUsage.htm

https://ifc43-docs.standards.buildingsmart.org/IFC/RELEASE/IFC4x3/HTML/lexical/IfcMaterialProfileSetUsage.htm